src/NClient/NClient.Standalone/ClientProxy/Generation/Interceptors/ClientInterceptor.cs

```diff
@@ -148,32 +153,58 @@ internal class ClientInterceptor<TClient, TRequest, TResponse> : AsyncIntercepto
             }
         }
         
-        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, CancellationToken cancellationToken)
+        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, ICachingAttribute? cachingAttribute = default, CancellationToken cancellationToken = default)
         {
+            if (await _responseCacheWorker.TryGet(request, cancellationToken) is { } cachedResult)
+            {
+                _toolset.Logger?.LogDebug("Response received from cache. Request id: '{requestId}'.", request.Id);
+                return cachedResult;
+            }
+            
             if (resultType == typeof(TResponse))
                 return await transportNClient
```

Comment:
We need to think about how to cache `TResponse`

Reply:
Yes